h2o-r/h2o-package/R/models.R

```diff
@@ -258,19 +266,34 @@ h2o.getFutureModel <- function(object,verbose=FALSE) {
     stop("Cannot find type ", paramDef$type, " in .type.map")
   if (scalar) { # scalar == TRUE
     if (type == "H2OModel")
-        type <-  "character"
-    if (!inherits(paramValue, type)) {
-      e <- paste0("\"", name , "\" must be of type ", type, ", but got ", class(paramValue), ".\n")
-    } else if ((length(paramDef$values) > 1L) && !(paramValue %in% paramDef$values)) {
-      e <- paste0("\"", name,"\" must be in")
-      for (fact in paramDef$values)
-        e <- paste0(e, " \"", fact, "\",")
-      e <- paste(e, "but got", paramValue)
+      type <-  "character"
+      if (name == "seed") {
+        if(is.character(paramValue) && !.is.int64(paramValue))
+          e <- paste0("\"seed\" must be of type long or string long, but got a string which cannot be converted to long.\n")
+        else if(is.numeric(paramValue)){
+          if(!.is.int64(paramValue)){
+            e <- paste0("\"seed\" must be of type long or string long, but got a number which cannot be converted to long.\n")
+          } else if(nchar(toString(paramValue)) > 16) {
```

Comment:
I don't think this works:

    > paramValue <- 100000000000000000
    > toString(paramValue)
    [1] "1e+17"

I tried this piece of code with the given paramValue and no warning was produced